_posts/2020-11-08-getting-back-into-headless-cloud-computing.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title:  "Getting Back Into Headless Cloud Computing"
+date:   2020-11-08 6:54:04 -0700
+categories: bash ssh tmux vpn headless
+---
+## Getting Back Into Headless Cloud Computing
+#### Use Case
+This tutorial answers the question, "How do I log back into a remote server and analyze data without a graphical user interface, again?"
+
+This is adapted from some notes I used when I was several months into a bioinformatics class and wanted to pick back up where I'd left off in the headless cloud environment.
+
+Since not all setups will be identical, and having several tools setup already is a prerequisite for this workflow, see the bottom of this post for my dependencies.
+
+#### Login
+* Open VPN app on your PC, e.g. Global Protect
+ * Enter your institutional account's username and password
+* Open Git Bash and initialize SSH connection
+ * Provide server-specific username and password, given to you by your instructor or IT admin for cloud computing
+ * e.g. `ssh Last_Name_Example@server_name.insitution.edu -p port_number`; enter provided password
+
+#### Shorten bash prompt
+ * Often, the bash prompt becomes unwieldy in length
+ * Try shortening it to just user and path: `PS1='\u:\W\$ '`
+ * Or, just path: `PS1='\W\$ '`
+
+#### Configure git in container
+* Find the right folder: `~`
+* Clone `git clone https://github.com/user/repo`
+* Set user `git config --global user.name “Full Name”`
+* Set email `git config --global user.email “preferred@email.com”`
+* Create new branch `git branch example-branch`
+* Switch to new branch `git checkout example-branch`
+* Add something, e.g. README `git add README.md`
+* Commit `git commit -m “Initialize example branch”`
+* Uncommit `git reset --soft HEAD~1`
+* Push `git push`
+* Enter GitHub username and password
+* Check `git config --list`
+
+#### Restore git session from different machine
+* Pull down branches made elsewhere `git pull`
```

Comment:
maybe mentions the options for like `origin master` etc

Reply:
Great, thanks, I've added this!

Reply:
@SpaceGold did you forget to push? it doesn't look like there have been any new commits after @melt6457's feedback

Reply:
Had to do some config stuff. Just pushed nominally final version with feedback!